### DISCUSSION TITLE
Tables in Grafana Table visualization can't be selected to copy-paste as table or individual cells

### DISCUSSION THREAD

**Discussion by MurzNN** [grafana/grafana#45146] [Dashboards, Panels & Visualizations]
**What happened**:
When data is represented in table mode, very often users work with them as table, not as regular text. So it's very common action to select several individual cells to copy them to clipboard, that actively used with tabled data on webpages.

**What you expected to happen**:
But with Grafana Panels in Table visualization mode we can't select individual cells, because they are represented as bunch of `div`s, instead of real HTML table.

With regular HTML table in Mozilla Firefox browser we can easily select individual cells or even cell range via holding the `Ctrl` button, also there are many extensions that helps copy (extract) data from tables: https://addons.mozilla.org/en-GB/firefox/search/?q=table%20copy

In Chromium and Blink engines you can use many extensions: https://chrome.google.com/webstore/search/table%20cell%20copy?_category=extensions.

And with Grafana Table we can select table cells only as regular text, that are pasted as text paragraphs for each cell.

So it's impossible to copy-paste Grafana Table output to some external target as table.

**How to reproduce it (as minimally and precisely as possible)**:
In Firefox - try to hold `Ctrl` and select cells range: 2 columns in 2-4 rows of 8x8 table, copy to clipboard, and paste them into MS Excel document or other WYSIWYG editor (CKEditor, etc).

In Chromium-based browsers - select several rows as plaintext and paste to CKEditor - they should be pasted as table cells, but will paste as separate text paragraph for each cell.

Now it isn't possible, you can select only the whole row, like plain text, not the individual cells.

**Environment**:
- Grafana version: v8.3.4

**Solution:**

The solution is to render tables as native HTML table, using React Table component or some other solution.

Here https://react-table.tanstack.com/docs/examples/bootstrap-ui-components you can try live examples to test functionality of selecting individual cells and columns in tables, copying them to clipboard and pasting to MS Word, Excel, CKEditor and other editors with table support.

Alternative solution can be add functionality to export tabled panel data as csv, but solution to switch to use native HTML tables should me much more easier and convenient.

**Comment by zoltanbedi** [comment]
Hi @MurzNN,
You can download the csv data see https://grafana.com/docs/grafana/latest/panels/inspect-panel/#download-raw-query-results-as-csv. Using table elements makes sense to me but we probably won't pick this up soon. We will consider this if more people show interest.

**Comment by FalconerTC** [comment]
I run into this problem often. Tables aren't particularly useful for me if I can't copy data from them for cross-correlation. Would love to see this feature!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by MurzNN** [comment]
Issue https://github.com/grafana/grafana/issues/82498
